File: src/abstractions/Bakabase.Abstractions/Models/Domain/Constants/RemoteDevicePlatform.cs
namespace Bakabase.Abstractions.Models.Domain.Constants;

/// <summary>
/// The OS family of a paired remote device. Drives which native-player hand-off
/// links the UI offers for that device.
/// </summary>
public enum RemoteDevicePlatform
{
    Unknown = 0,
    Windows = 1,
    MacOS = 2,
    Linux = 3,
    Android = 4,
    IOS = 5
}

File: src/web/src/core/remoteDevicePlatform.ts
import { RemoteDevicePlatform } from "@/sdk/constants";

/**
 * The label for the platform a paired device reported. Shared by the remote-access
 * settings and the devices page, which list the same devices.
 */
export const remoteDevicePlatformLabelKey = (platform?: RemoteDevicePlatform) => {
  switch (platform) {
    case RemoteDevicePlatform.Windows:
      return "configuration.remoteAccess.platform.windows";
    case RemoteDevicePlatform.MacOS:
      return "configuration.remoteAccess.platform.macOS";
    case RemoteDevicePlatform.Linux:
      return "configuration.remoteAccess.platform.linux";
    case RemoteDevicePlatform.Android:
      return "configuration.remoteAccess.platform.android";
    case RemoteDevicePlatform.IOS:
      return "configuration.remoteAccess.platform.iOS";
    default:
      return "configuration.remoteAccess.platform.unknown";
  }
};
